### PR TITLE
Task/stad 367 measurement duration

### PR DESCRIPTION
**Pull request by hb0**
This PR adds:
- average speed calculation
- measurement duration calculation

**Recommendation:**
Review PR via "Files changed" or else you'll review much more code then necessary (average speed stored in database which was reverted).

### CODE REVIEW

File: persistence/build.gradle
Comment (hb0) [inline persistence/build.gradle:51]: Without this, we would need to fix ~ 30 lint errors here:

```
ret += deleteDataForMeasurement(database, selectedMeasurementsCursor
                                        .getLong(selectedMeasurementsCursor.getColumnIndex(BaseColumns._ID)));
```

=> getColumnIndex can be -1 when the column does not exist, but `cursor.getLong()` only accepts an index int > 0

I'm not sure if we want to do much "cleanup" work in the SDK 6 fork, so I added the "ignore lint" option instead.

**Review by hb0 — COMMENTED** [review]
pre-reviewed

File: persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java
Comment (hb0) [inline persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java:589]: I think it's expected, that the average speed does not decrease when the user is not moving for a short time. (also, because the customer said "should be like distance calculation)
Thus, I'm adding speed samples only after the location cleaning strategy accepts the location captured.

My understanding of the feature request was:

the average speed per measurement should be calculated
i.e. when we pause the first track of the measurement and resume, the speed samples of all sub-tracks are used to calculate the average speed
for this to work, we cannot just save "averageSpeed" but have to save the speed sum and counter.

**Review by hb0 — COMMENTED** [review]
pre-reviewed

File: persistence/build.gradle
Comment (muthenberg) [reply persistence/build.gradle:51]: Yeah. Just do not merge that to the main branch in the future.

File: persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java
Comment (muthenberg) [reply persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java:589]: The customer did not say anything about "distance calculation". That is what I said. We should probably have a look into how other apps do this, before making a final decision.

**Review by muthenberg — CHANGES_REQUESTED** [review]
We need to have a look into another Bike Statistics App like Komoot, Strava, Runtastic or the like, before deciding on whether to include pauses while averaging. I am not sure how this should be implemented.

File: persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java
Comment (hb0) [reply persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java:589]: see discussion in Ticket

Comment (muthenberg) [reply persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java:589]: Seems there is no easy answer to this. Lets got with this implementation then.

**Review by muthenberg — APPROVED** [review]

**Pull request merged**
